parser_library/src/processing/opencode_provider.cpp

```diff
@@ -193,22 +193,48 @@ void opencode_provider::ainsert(const std::string& rec, ainsert_destination dest
     suspend_copy_processing(remove_empty::no);
 }
 
-void opencode_provider::feed_line(const parsing::parser_holder& p, bool is_process)
+namespace {
+void produce_hl_symbols(
+    const lexing::logical_line<utils::utf8_iterator<std::string_view::iterator, utils::utf8_utf16_counter>>& ll,
+    size_t lineno,
+    semantics::source_info_processor& sip)
+{
+    static constexpr auto ll_range = [](size_t lno, const auto& b, const auto& m, const auto& e) {
+        return range({ lno, lexing::logical_distance(b, m) }, { lno, lexing::logical_distance(b, e) });
```

Comment:
Would it be better to have the following in case `lexing::logical_distance` doesn't support random access?

```suggestion
        auto start = lexing::logical_distance(b, m);
        return range({ lno, start}, { lno, start + lexing::logical_distance(m, e) });
```